flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/dataformat/TypeGetterSetters.java

```diff
@@ -205,8 +216,45 @@ static Object get(TypeGetterSetters row, int ordinal, InternalType type) {
 			return row.getRow(ordinal, ((RowType) type).getArity());
 		} else if (type instanceof GenericType) {
 			return row.getGeneric(ordinal);
+		} else if (type.equals(InternalTypes.BINARY)) {
+			return row.getBinary(ordinal);
 		} else {
 			throw new RuntimeException("Not support type: " + type);
 		}
 	}
+
+	/**
+	 * Get binary, if len less than 8, will be include in variablePartOffsetAndLen.
+	 *
+	 * <p>If len is less than 8, its binary format is:
+	 * 1bit mark(1) = 1, 7bits len, and 7bytes data.
+	 *
+	 * <p>If len is greater or equal to 8, its binary format is:
+	 * 1bit mark(1) = 0, 31bits offset, and 4bytes len.
+	 * Data is stored in variable-length part.
+	 *
+	 * <p>Note: Need to consider the ByteOrder.
+	 *
+	 * @param baseOffset base offset of composite binary format.
+	 * @param fieldOffset absolute start offset of 'variablePartOffsetAndLen'.
+	 * @param variablePartOffsetAndLen a long value, real data or offset and len.
+	 */
+	static byte[] readBinaryFieldFromSegments(
```

Comment:
Do we have any better place to put utility function like this?

Reply:
BinaryFormat is better.